02b_widgets.ipynb

```diff
@@ -539,118 +656,204 @@
     "\n",
-    "def display_selected_data(change): \n",
-    "    selected_data_output.clear_output(wait=True)\n",
-    "    with selected_data_output: \n",
-    "        display(selected)"
+    "# Call the function (since called outside observe method, use None as argument)\n",
+    "output_selected_dataframe(None)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8f2cc082-3d7f-4c2f-9efd-2537d39a851a",
+   "metadata": {},
+   "source": [
+    "Cool! Now the only thing that is missing is that we want to call this function every time the range is updated. How might we go about that?\n",
+    "\n",
+    "Let's connect the `output_selected_dataframe` function to the `year_range` widget with the `observe` method."
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "b1ea50be-0978-431d-a4a0-9f99429f6ac1",
+   "id": "1f83030d-648e-4ad3-99bb-8835fa07c2fb",
    "metadata": {},
    "outputs": [],
    "source": [
-    "#| export\n",
-    "display_selected_data(None)"
+    "year_range.observe(output_selected_dataframe, 'value')"
    ]
   },
   {
    "cell_type": "markdown",
-   "id": "8f2cc082-3d7f-4c2f-9efd-2537d39a851a",
+   "id": "c47a2de8-6193-40bb-8f26-f33107b5099c",
+   "metadata": {},
+   "source": [
+    "Okay let's create another copy of the slider so we can change the value. Then go ahead and move the range around and see if you notice changes to the displayed dataframe above."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e7647b5d-b4d4-46a9-b1a3-8bd626a1a4d4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "year_range"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7510f842",
    "metadata": {},
    "source": [
-    "Cool! Now the only thing that is missing is that we want to call this function every time the range is updated. How might we go about that?"
+    "Way cool. Now our dataframe reflects the range of values."
    ]
   },
   {
    "cell_type": "markdown",
-   "id": "57866615-000c-4be9-b551-821e9f36829c",
+   "id": "eab31b81-150d-4065-add3-e2460bfc63b1",
    "metadata": {},
    "source": [
-    "Well, we could use another observe function! Let's try it."
+    "### A Better Way to Display Dataframes\n",
+    "\n",
+    "Notice that the size of the displayed pandas dataframe changes because default display of a dataframe is to only show up to 30 rows. This means our Output widget changes size when we change the range of years in the selected data.\n",
+    "\n",
+    "Another approach is rather than using the generic output widget to instead use one that is specifically designed for pandas dataframes. It's called [ipydatagrid](https://github.com/bloomberg/ipydatagrid). Let's import it and see how it works."
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "1f83030d-648e-4ad3-99bb-8835fa07c2fb",
+   "id": "ae5f9cca",
    "metadata": {},
    "outputs": [],
    "source": [
     "#| export\n",
```

Comment:
There is an important error here -- `nbdev` cannot handle an import and other code in the same cell, so this needs to be broken into two `#| export` cells, on for the import and one for creating `selector_data_grid`

Reply:
Never mind, starting with a clesn checkout worked.